.gitlab-ci.yml

```diff
@@ -1,10 +1,66 @@
-# other GitLab CI YAML files to pull jobs and job templates from -- note that there is an
-# additional "forest" directory here, as the GitLab version of the repo is used by include
-include:
-  - project: rigetti/forest/gitlab-pipelines
-    file: docker.gitlab-ci.yml
-  - project: rigetti/forest/gitlab-pipelines
-    file: python.gitlab-ci.yml
+default:
+  image: python:3.7
+  tags:
+    - ec2-docker
+
+stages:
+  - test
+  - deploy
+  - docker
+
+cache:
+  key: "${CI_COMMIT_REF_SLUG}"
+  paths:
+    - .cache/pip
+    - .venv
+
+.install-dependencies:
+  &install-dependencies # Recommended install method for Poetry: https://python-poetry.org/docs/
+  - curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+  - source $HOME/.poetry/env
+  - poetry --version
+  - poetry config virtualenvs.in-project true
+  - poetry config experimental.new-installer false  # Address flaky poetry installs, see: https://github.com/python-poetry/poetry/issues/3199
+  - poetry run python -m ensurepip --upgrade
```

Comment:
This was really useful to get around a strange issue where pip wasn't present (only occurred in the python 3.7 tests, no issues for 3.8 🤷 )